### ISSUE TITLE
do not show error in symbolic link folder

### ISSUE THREAD

**Issue by zhuoyan123** [microsoft/vscode-eslint#1351]
I worked on mac os, this is my workspace
![20211013092038](https://user-images.githubusercontent.com/43226113/137050507-b87a1889-d741-4c4b-83f2-27f76331155a.jpg)

the 'src' is linked to another folder, when I make error in some file in src, vscode-eslint can not give me error info, but webpack gives me.

this is stat when I maked an error
![image](https://user-images.githubusercontent.com/43226113/137051246-b4d8af5c-56d9-4e2b-9b58-b659e9b52673.png)

this is the info by command 'ls -l' in root
![image](https://user-images.githubusercontent.com/43226113/137050804-b5c2d260-cba1-4101-bcea-e94258b7aae1.png)

**Comment by dbaeumer** [comment]
All I can recommend is to exclude the folder in this case using an `.eslintignore` file

**Comment by dbaeumer** [comment]
The validation doesn't happen correctly in the terminal. If I run `./node_modules/.bin/eslint src/App.test.js` I get

```
Oops! Something went wrong! :(

ESLint: 7.11.0

ESLint couldn't find the plugin "eslint-plugin-flowtype".

(The package "eslint-plugin-flowtype" was not found when loaded as a Node module from the directory "/workspaces/vscode-eslint-symlinks/project".)

It's likely that the plugin isn't installed correctly. Try reinstalling by running the following:

    npm install eslint-plugin-flowtype@latest --save-dev

The plugin "eslint-plugin-flowtype" was referenced from the config file in ".eslintrc.js » eslint-config-react-app".

If you still can't figure out the problem, please stop by https://eslint.org/chat/help to chat with the team.
```

The extension can only validate files correctly iff the validation happens correctly in the terminal. I will close the issue for now. 

If you still see problems after the validation happens correctly in the terminal please ping and I will reopen.

**Comment by CarstenWalther** [comment]
I had the same problem. The eslint extension did not show linter errors from files in a symlinked folder. Running eslint from the terminal worked fine though.

As a workaround set the `nodePath` configuration to your node_modules directory. E.g. `"eslint.nodePath": "node_modules"`. Then it should show linter errors for files in symlinked folders.

**Comment by romamik** [comment]
Can confirm that issue is still here and that the workaround suggested by @CarstenWalther works.
I've prepared the repository with reproduction: https://github.com/romamik/eslint-vs-code-bug
There is a readme with steps to reproduce there.
@dbaeumer  Please consider reopening this issue.

**Comment by dbaeumer** [comment]
@romamik thanks for the GitHub repository. The underlying problem here is how node deals with sym links and there is nothing I can do about this except that you can provide me with the location of the eslint module by settting `eslint.nodePath`. Here is what is happening in node:

```
@dbaeumer ➜ /workspaces/eslint-vs-code-bug/vscode-workspace/src-symlinked (master) $ node
Welcome to Node.js v16.18.1.
Type ".help" for more information.
> process.cwd()
'/workspaces/eslint-vs-code-bug/my-src'
```

Even though I start node on the symlinked folder node does a realpath and actually opens on `'/workspaces/eslint-vs-code-bug/my-src'`. As a consequence, the eslint server doesn't find an Eslint npm package since there is none up the parent chain from `'/workspaces/eslint-vs-code-bug/my-src'`. 

I tried various things to convince node to not fall back to the real path but wasn't able to do so. If someone has an idea I am happy to further experiement.

It does work in the terminal since there we start with a symlinked path in the terminal.

**Comment by dbaeumer** [comment]
An easier fix is to denote the workspace root as the working directory to use by using the following setting:

```
	"eslint.workingDirectories": [
		"."
	]
```

**Comment by romamik** [comment]
They are speaking about __filename/__dirname there, but they behave the same. https://github.com/nodejs/node/issues/22602

I agree that nothing can be done here without proper information from the node.
